File: demo_python_tf/demo_python_tf/tf_listener.py
import rclpy
from rclpy.node import Node
from tf2_ros import TransformListener,Buffer,TransformBroadcaster
from geometry_msgs.msg import TransformStamped
from tf_transformations import euler_from_quaternion
import math 


class TFBroadcaster(Node):
    
    def __init__(self):
        super().__init__('tf_broadcaster')
        self.buffer_ = Buffer()
        self.listener_ = TransformListener(self.buffer_,self)
        self.timer_ = self.create_timer(1.0,self.get_transform)
        
    
    def get_transform(self):
        
        """
        get current coordinate relationship
        """
        
        try:
            result = self.buffer_.lookup_transform('bottom_link','camera_link',rclpy.time.Time(seconds=0),rclpy.time.Duration(seconds=1))
            transform = result.transform
            self.get_logger().info(f"translate : {transform.translation}")
            self.get_logger().info(f"translate : {transform.rotation}")
            
            rotation_euler = euler_from_quaternion([
                transform.rotation.x,
                transform.rotation.y,
                transform.rotation.z,
                transform.rotation.w]
            )
            self.get_logger().info(f"rotation RPY : {rotation_euler}")
            
           
        except Exception as e:
            self.get_logger().warn(f"fail to get coordiate transformation with error :{str(e)}")
            
    
        

def main():
    
    rclpy.init()
    node = TFBroadcaster()
    rclpy.spin(node)
    rclpy.shutdown()
    

File: demo_cpp_tf/src/tf_listener.cpp
#include "geometry_msgs/msg/transform_stamped.hpp"
#include "rclcpp/rclcpp.hpp"
#include "tf2/LinearMath/Quaternion.h"
#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
#include "tf2_ros/transform_listener.h"
#include "tf2_ros/buffer.h" 
#include "chrono"
#include "tf2/utils.h"

using namespace std::chrono_literals;

class TFListener: public rclcpp::Node
{
private:
   std::shared_ptr<tf2_ros::TransformListener> listener_;
   rclcpp::TimerBase::SharedPtr timer_;
   std::shared_ptr<tf2_ros::Buffer> buffer_;

public:
    TFListener():Node("tf_listener"){
        this->buffer_ = std::make_shared<tf2_ros::Buffer>(this->get_clock());
        listener_ = std::make_shared<tf2_ros::TransformListener>(*buffer_,this);
        timer_ = this->create_wall_timer(100ms,std::bind(&TFListener::getTransform,this));
        
    }
    
    void getTransform(){
        try
        {
            const auto transform_ = buffer_->lookupTransform(
                "base_link","target_point",this->get_clock()->now(),rclcpp::Duration::from_seconds(1));
            auto translation = transform_.transform.translation;
            auto rotation = transform_.transform.rotation;
            double y,p,r;

            tf2::getEulerYPR(rotation,y,p,r);
            RCLCPP_INFO(get_logger(),"translate : %f, %f, %f",translation.x,translation.y,translation.z);
            RCLCPP_INFO(get_logger(),"rotate : %f, %f, %f",y,p,r);
            }
        catch(const std::exception& e)
        {
           RCLCPP_WARN(get_logger(),"%s",e.what());
        }
        
    }
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<TFListener>();
    rclcpp::spin(node);
    return 0;
}

